inflate_concurrent_test.go

```
@@ -0,0 +1,194 @@
+	// check runs one decode into a freshly allocated destination and
+	// reports any deviation from the precomputed expectation. It must stay
+	// t.Fatal-free: it runs on non-test goroutines.
+	check := func(c *inflateCase) error {
+		dst := make([]byte, c.dstSize)
+		consumed, err := inflatePackZlibGo(c.src, dst)
+		if c.wantErr {
+			if err == nil {
+				return fmt.Errorf("%s: malformed stream accepted (consumed=%d)", c.name, consumed)
+			}
+			if got := errors.Is(err, io.ErrUnexpectedEOF); got != c.wantEOF {
+				return fmt.Errorf("%s: error %v has EOF class %v, want %v", c.name, err, got, c.wantEOF)
```

golangci-lint (2.12.2)

[error] non-wrapping format verb for fmt.Errorf. Use `%w` to format errors

(errorlint)